makefiles/buildtest: always execute 'buildtest' loop on host machine

cladmi opened the pull request:
### Contribution description 

This remove executing buildtest `for` loop in docker.

When building completely in docker, 'buildtest' would hide issues when
the host toolchain would be used when doing `make all` directly.

A side effect is also that now `BUILDTEST_MAKE_REDIRECT` would work when
doing `buildtest` with docker.
This also now removes passing `BOARDS` as it was added and needed only
for 'buildtest', and should work without the specific handling. https://github.com/RIOT-OS/RIOT/pull/11385


The side effect is that now one `docker` instance should be started per compilation but makes it more consistent with  doing `BUILD_IN_DOCKER=1 BOARD=board_name make` for each board.

### Testing procedure

I locally used `DOCKER="sudo docker"` but did not include it in the tests commands displayed below.


#### Normal buildtest uses

Running `buildtest` with `BUILD_IN_DOCKER=1` still works.

```
BUILD_IN_DOCKER=1 make --no-print-directory -C examples/hello-world/ buildtest
Building for acd52832 ... success.
Building for airfy-beacon ... success.
...
Building for wsn430-v1_3b ... success.
Building for wsn430-v1_4 ... success.
Building for yunjia-nrf51822 ... success.
Building for z1 ... success.
```

`BOARDS` can still be limited:

```
BUILD_IN_DOCKER=1 BOARDS='native samr21-xpro pic32-wifire' make --no-print-directory -C examples/hello-world/ buildtest
Building for native ... success.
Building for samr21-xpro ... success.
Building for pic32-wifire ... success.
```

When setting `BUILDTEST_MAKE_REDIRECT=` in the environment it will be taken into account

<details><summary><code>BUILDTEST_MAKE_REDIRECT= BUILD_IN_DOCKER=1 BOARDS='native samr21-xpro' make --no-print-directory -C examples/default buildtest</code></summary>

```
BUILDTEST_MAKE_REDIRECT= BUILD_IN_DOCKER=1 BOARDS='native samr21-xpro' make --no-print-directory -C examples/default buildtest
Building for native ... Launching build container using image "riot/riotbuild:latest".
sudo docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Berlin:/etc/localtime:ro' -v '/home/harter/work/git/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles' -v /home/harter/.gitcache:/data/riotbuild/gitcache -e GIT_CACHE_DIR=/data/riotbuild/gitcache   \
    -e 'BOARD=native' -e 'RIOT_CI_BUILD=1' \
    -w '/data/riotbuild/riotbase/examples/default/' \
    'riot/riotbuild:latest' make all
Building application "default" for "native" with MCU "native".

   text    data     bss     dec     hex filename
  90210    1104   72024  163338   27e0a /data/riotbuild/riotbase/examples/default/bin/native/default.elf
success.
Building for samr21-xpro ... Launching build container using image "riot/riotbuild:latest".
sudo docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Berlin:/etc/localtime:ro' -v '/home/harter/work/git/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles' -v /home/harter/.gitcache:/data/riotbuild/gitcache -e GIT_CACHE_DIR=/data/riotbuild/gitcache   \
    -e 'BOARD=samr21-xpro' -e 'RIOT_CI_BUILD=1' \
    -w '/data/riotbuild/riotbase/examples/default/' \
    'riot/riotbuild:latest' make all
Building application "default" for "samr21-xpro" with MCU "samd21".

   text    data     bss     dec     hex filename
  40856     508    6048   47412    b934 /data/riotbuild/riotbase/examples/default/bin/samr21-xpro/default.elf
success.
```
</details>


In master it was showing the `docker` command first to run `buildtest` now `all` is run with `docker` for each board.

<details><summary>Master: <code>BUILDTEST_MAKE_REDIRECT= BUILD_IN_DOCKER=1 BOARDS='native samr21-xpro' make --no-print-directory -C examples/hello-world/ buildtest</code></summary>

```
BUILDTEST_MAKE_REDIRECT= BUILD_IN_DOCKER=1 BOARDS='native samr21-xpro' make --no-print-directory -C examples/hello-world/ buildtest
Launching build container using image "riot/riotbuild:latest".
mkdir -p /home/harter/work/git/worktree/riot_master/build
sudo docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Berlin:/etc/localtime:ro' -v '/home/harter/work/git/worktree/riot_master:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles' -v /home/harter/.gitcache:/data/riotbuild/gitcache -e GIT_CACHE_DIR=/data/riotbuild/gitcache  -v /home/harter/work/git/RIOT/.git:/home/harter/work/git/RIOT/.git \
    -e 'BOARDS=native samr21-xpro' \
    -w '/data/riotbuild/riotbase/examples/hello-world/' \
    'riot/riotbuild:latest' make buildtest  
Building for native ... success.
Building for samr21-xpro ... success.
```
</details>

#### Removes errors hiding

Use a PATH where you do not have `arm-none-eabi-gcc`
```
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin arm-none-eabi-gcc

Command 'arm-none-eabi-gcc' not found, but can be installed with:

sudo apt install gcc-arm-none-eabi
```

It now correctly fails with `buildtest` with `BUILD_IN_DOCKER=1` as the host toolchain is used by that application.


<details><summary><code>BOARDS="iotlab-m3 samr21-xpro" PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/riotboot buildtest</code></summary>

```
BOARDS="iotlab-m3 samr21-xpro" PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/riotboot buildtest
make: Entering directory '/home/harter/work/git/RIOT/tests/riotboot'
Building for iotlab-m3 ... /bin/sh: 1: arm-none-eabi-gcc: not found
make[1]: *** [/home/harter/work/git/RIOT/tests/riotboot/bin/iotlab-m3/tests_riotboot-slot0.elf] Error 127
failed!
Building for samr21-xpro ... /bin/sh: 1: arm-none-eabi-gcc: not found
make[1]: *** [/home/harter/work/git/RIOT/tests/riotboot/bin/samr21-xpro/tests_riotboot-slot0.elf] Error 127
failed!
/home/harter/work/git/RIOT/makefiles/buildtests.inc.mk:6: recipe for target 'buildtest' failed
make: *** [buildtest] Error 1
make: Leaving directory '/home/harter/work/git/RIOT/tests/riotboot'
```

</details>

When it was hiding the error and correctly compiling in `master`

<details><summary><code>DOCKER="sudo docker" BOARDS="iotlab-m3 samr21-xpro" PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/riotboot buildtest</code></summary>

```
DOCKER="sudo docker" BOARDS="iotlab-m3 samr21-xpro" PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/riotboot buildtest
make: Entering directory '/home/harter/work/git/RIOT/tests/riotboot'
Launching build container using image "riot/riotbuild:latest".
mkdir -p /home/harter/work/git/RIOT/build
sudo docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Berlin:/etc/localtime:ro' -v '/home/harter/work/git/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles' -v /home/harter/.gitcache:/data/riotbuild/gitcache -e GIT_CACHE_DIR=/data/riotbuild/gitcache   \
    -e 'BOARDS=iotlab-m3 samr21-xpro' \
    -w '/data/riotbuild/riotbase/tests/riotboot/' \
    'riot/riotbuild:latest' make buildtest
Building for iotlab-m3 ... «success.
Building for samr21-xpro ... success.
make: Leaving directory '/home/harter/work/git/RIOT/tests/riotboot'
```
</details>

and also hiding the error and correctly compiling in with `buildtest-indocker`.

<details><summary><code>BOARDS="iotlab-m3 samr21-xpro" PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/riotboot buildtest-indocker</code></summary>
```
BOARDS="iotlab-m3 samr21-xpro" PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/riotboot buildtest-indocker
make: Entering directory '/home/harter/work/git/RIOT/tests/riotboot'
Launching build container using image "riot/riotbuild:latest".
mkdir -p /home/harter/work/git/RIOT/build
sudo docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Berlin:/etc/localtime:ro' -v '/home/harter/work/git/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles' -v /home/harter/.gitcache:/data/riotbuild/gitcache -e GIT_CACHE_DIR=/data/riotbuild/gitcache   \
    -e 'BOARDS=iotlab-m3 samr21-xpro' \
    -w '/data/riotbuild/riotbase/tests/riotboot/' \
    'riot/riotbuild:latest' make buildtest-indocker
Building for iotlab-m3 ... success.
Building for samr21-xpro ... success.
make: Leaving directory '/home/harter/work/git/RIOT/tests/riotboot'
```
</details>

<details><summary>mcuboot is now working since https://github.com/RIOT-OS/RIOT/pull/11083</summary>

<details><summary><code>PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/mcuboot/ buildtest</code></summary>

```
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/mcuboot/ buildtest
make: Entering directory '/home/harter/work/git/RIOT/tests/mcuboot'
Building for nrf52dk ... /bin/sh: 1: arm-none-eabi-gcc: not found
make[1]: *** [mcuboot] Error 127
failed!
/home/harter/work/git/RIOT/makefiles/buildtests.inc.mk:6: recipe for target 'buildtest' failed
make: *** [buildtest] Error 1
make: Leaving directory '/home/harter/work/git/RIOT/tests/mcuboot'
```
</details>

When it was hiding the error and correctly compiling in `master`

<details><summary><code>PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 make -C tests/mcuboot/ buildtest</code></summary>

```
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 make -C tests/mcuboot/ buildtest
make: Entering directory '/home/harter/work/git/worktree/riot_master/tests/mcuboot'
Launching build container using image "riot/riotbuild:latest".
mkdir -p /home/harter/work/git/worktree/riot_master/build
sudo docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Berlin:/etc/localtime:ro' -v '/home/harter/work/git/worktree/riot_master:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles' -v /home/harter/.gitcache:/data/riotbuild/gitcache -e GIT_CACHE_DIR=/data/riotbuild/gitcache  -v /home/harter/work/git/RIOT/.git:/home/harter/work/git/RIOT/.git \
     \
    -w '/data/riotbuild/riotbase/tests/mcuboot/' \
    'riot/riotbuild:latest' make buildtest  
Building for nrf52dk ... success.
make: Leaving directory '/home/harter/work/git/worktree/riot_master/tests/mcuboot'
```
</details>

</details>


### Issues/PRs references

Found that `BUILD_IN_DOCKER=1 make buildtest` was hiding expected errors during 2019.07-RC1 testing https://github.com/RIOT-OS/Release-Specs/issues/128#issuecomment-511388172

Fixes https://github.com/RIOT-OS/RIOT/issues/11842

fjmolinas commented:
Does build time increase in a significant way because docker is started per compilation?

cladmi commented:
It does indeed increase I will do a measure on my machine.

However, if not changing it because of execution time, I would see it as a "disable test as it takes more time".
This shows issues that happen on a developer machine that murdock does not show as it is always inside the container and never with `BUILD_IN_DOCKER=1`.

cladmi commented:
The time increase is indeed significant. It changes from 10 to 20 minutes for `examples/hello-world`.

Should it be split in two different targets? So `buildtest` would be the real test and a `buildtest-no-local-toolchain-check` something ?

fjmolinas commented:
There is something I don't quite understand here, why should building in `docker` use the host tool chain in the first place?

cladmi commented:
It MUST not use the host toolchain. That is why when testing building with `docker` on my test machine I do not have any local toolchain.
(except native gcc as it is required for flashing for kinetis, or compiling host 'edbg')

However, the `BUILD_IN_DOCKER` integration is based on an internal hook that calls make again inside a prepared docker container. For a normal compilation it is through this, that then is handled in `makefiles/docker.inc.mk`:

https://github.com/RIOT-OS/RIOT/blob/950b83e63e8310ac1e94be360d826ea701bee60e/Makefile.include#L460-L467

But some applications, are currently handled in a way that the build system still executes targets outside of this handling:

* `tests/mcuboot` https://github.com/RIOT-OS/RIOT/pull/11083
* `tests/riotboot`, `bootloader/riotboot` https://github.com/RIOT-OS/RIOT/issues/12003


#### Consequence on `buildtest`

With the current way `buildtest` is implemented with `BUILD_IN_DOCKER`, the compilation command:

https://github.com/RIOT-OS/RIOT/blob/950b83e63e8310ac1e94be360d826ea701bee60e/makefiles/buildtests.inc.mk#L14-L15 

is executed from inside the container, so without `BUILD_IN_DOCKER=1`.
Because of this, it never uses the handling in `Makefile.include` directly on your machine.
And this hides the issues you would have when doing `BUILD_IN_DOCKER=1 make all` and that I would like to have visible.

fjmolinas commented:
@cladmi Ok now I understand the issue correctly. 

> So buildtest would be the real test and a buildtest-no-local-toolchain-check something ?

+1 for this approach, all tough I would rename it to `buildtests-indocker` or something like that. Would murdock need to be changed to avoid the x2 in build time? I think it's good enough if during release testing we use just `build-test` and  `buildtests-indocker` normally. What do you think?

cladmi commented:
Murdock is not using the `BUILD_IN_DOCKER=1` at all, and not `buildtest` either. So no impact there.

It is only during release, or developers on their machine that would run `buildtest`.

Good for `buildtest-indocker`, I will update it.

cladmi commented:
PR needs squashing.

cladmi commented:
The testing prodecures from the first post are still valid and get the same output.

The new `buildtest-indocker` gives the same outpus as `buildtest` before with `BUILD_IN_DOCKER=1`:

<details><summary><code>BUILDTEST_MAKE_REDIRECT= BUILD_IN_DOCKER=1 DOCKER="sudo docker" BOARDS="native samr21-xpro" make --no-print-directory -C examples/default/ buildtest-indocker</code></summary>

```
BUILDTEST_MAKE_REDIRECT= BUILD_IN_DOCKER=1 DOCKER="sudo docker" BOARDS="native samr21-xpro" make --no-print-directory -C examples/default/ buildtest-indocker
Launching build container using image "riot/riotbuild:latest".
mkdir -p /home/harter/work/git/RIOT/build
sudo docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Berlin:/etc/localtime:ro' -v '/home/harter/work/git/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles' -v /home/harter/.gitcache:/data/riotbuild/gitcache -e GIT_CACHE_DIR=/data/riotbuild/gitcache   \
    -e 'BOARDS=native samr21-xpro' \
    -w '/data/riotbuild/riotbase/examples/default/' \
    'riot/riotbuild:latest' make buildtest-indocker
Building for native ... success.
Building for samr21-xpro ... success.
```
</details>

And when executed without `BUILD_IN_DOCKER=1` it displays an error:

```
make --no-print-directory -C examples/default/ buildtest-indocker
/home/harter/work/git/RIOT/makefiles/buildtests.inc.mk:35: *** buildtest-indocker must be run with `BUILD_IN_DOCKER=1`.  Stop.
```

cladmi commented:
I rebased now that https://github.com/RIOT-OS/RIOT/pull/11083 was merged and updated the testing procedure in the PR description to use `tests/riotboot`.

fjmolinas commented:
Verified test procedure.

<details><summary><b>master:</b> BUILDTEST_MAKE_REDIRECT= BUILD_IN_DOCKER=1 BOARDS='native samr21-xpro' make --no-print-directory -C examples/default buildtest</summary>

```
Launching build container using image "riot/riotbuild:latest".
mkdir -p /home/francisco/workspace/RIOT/build
docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Paris:/etc/localtime:ro' -v '/home/francisco/workspace/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'     \
    -e 'BOARDS=native samr21-xpro' \
    -w '/data/riotbuild/riotbase/examples/default/' \
    'riot/riotbuild:latest' make buildtest  
Building for native ... success.
Building for samr21-xpro ... success.


```
</details>

<details><summary><b>PR:</b> BUILDTEST_MAKE_REDIRECT= BUILD_IN_DOCKER=1 BOARDS='native samr21-xpro' make --no-print-directory -C examples/default buildtest</summary>

```
Building for native ... Launching build container using image "riot/riotbuild:latest".
docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Paris:/etc/localtime:ro' -v '/home/francisco/workspace/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'     \
    -e 'BOARD=native' -e 'BOARDS=native samr21-xpro' -e 'RIOT_CI_BUILD=1' \
    -w '/data/riotbuild/riotbase/examples/hello-world/' \
    'riot/riotbuild:latest' make all  
Building application "hello-world" for "native" with MCU "native".

   text	   data	    bss	    dec	    hex	filename
  22918	    568	  47652	  71138	  115e2	/data/riotbuild/riotbase/examples/hello-world/bin/native/hello-world.elf
success.
Building for samr21-xpro ... Launching build container using image "riot/riotbuild:latest".
docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Paris:/etc/localtime:ro' -v '/home/francisco/workspace/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbaseBuilding for native ... Launching build container using image "riot/riotbuild:latest".
docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Paris:/etc/localtime:ro' -v '/home/francisco/workspace/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'     \
    -e 'BOARD=native' -e 'BOARDS=native samr21-xpro' -e 'RIOT_CI_BUILD=1' \
    -w '/data/riotbuild/riotbase/examples/hello-world/' \
    'riot/riotbuild:latest' make all  
Building application "hello-world" for "native" with MCU "native".

   text	   data	    bss	    dec	    hex	filename
  22918	    568	  47652	  71138	  115e2	/data/riotbuild/riotbase/examples/hello-world/bin/native/hello-world.elf
success.
Building for samr21-xpro ... Launching build container using image "riot/riotbuild:latest".
docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Paris:/etc/localtime:ro' -v '/home/francisco/workspace/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'     \
    -e 'BOARD=samr21-xpro' -e 'BOARDS=native samr21-xpro' -e 'RIOT_CI_BUILD=1' \
    -w '/data/riotbuild/riotbase/examples/hello-world/' \
    'riot/riotbuild:latest' make all  
Building application "hello-world" for "samr21-xpro" with MCU "samd21".

   text	   data	    bss	    dec	    hex	filename
   8216	    120	   2548	  10884	   2a84	/data/riotbuild/riotbase/examples/hello-world/bin/samr21-xpro/hello-world.elf' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'     \
    -e 'BOARD=samr21-xpro' -e 'BOARDS=native samr21-xpro' -e 'RIOT_CI_BUILD=1' \
    -w '/data/riotbuild/riotbase/examples/hello-world/' \
    'riot/riotbuild:latest' make all  
Building application "hello-world" for "samr21-xpro" with MCU "samd21".

   text	   data	    bss	    dec	    hex	filename
   8216	    120	   2548	  10884	   2a84	/data/riotbuild/riotbase/examples/hello-world/bin/samr21-xpro/hello-world.elf

```
</details>

<details><summary> <b>master:</b>BOARDS="iotlab-m3 samr21-xpro" PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/riotboot buildtest
</summary>

```
Launching build container using image "riot/riotbuild:latest".
mkdir -p /home/francisco/workspace/RIOT/build
docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Paris:/etc/localtime:ro' -v '/home/francisco/workspace/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'     \
    -e 'BOARDS=iotlab-m3 samr21-xpro' \
    -w '/data/riotbuild/riotbase/tests/riotboot/' \
    'riot/riotbuild:latest' make buildtest  
Building for iotlab-m3 ... success.
Building for samr21-xpro ... success.
make: Leaving directory '/home/francisco/workspace/RIOT/tests/riotboot'

```
</details>

<details><summary> <b>PR:</b>BOARDS="iotlab-m3 samr21-xpro" PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/riotboot buildtest
</summary>

```
make: Entering directory '/home/francisco/workspace/RIOT/tests/riotboot'
Building for iotlab-m3 ... /bin/sh: 1: arm-none-eabi-gcc: not found
make[1]: *** [/home/francisco/workspace/RIOT/tests/riotboot/bin/iotlab-m3/tests_riotboot-slot0.elf] Error 127
failed!
Building for samr21-xpro ... /bin/sh: 1: arm-none-eabi-gcc: not found
make[1]: *** [/home/francisco/workspace/RIOT/tests/riotboot/bin/samr21-xpro/tests_riotboot-slot0.elf] Error 127
failed!
/home/francisco/workspace/RIOT/makefiles/buildtests.inc.mk:6: recipe for target 'buildtest' failed
make: *** [buildtest] Error 1
make: Leaving directory '/home/francisco/workspace/RIOT/tests/riotboot'

```
</details>

<details><summary> <b>PR:</b>BOARDS="iotlab-m3 samr21-xpro" PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin BUILD_IN_DOCKER=1 BUILDTEST_MAKE_REDIRECT='>/dev/null' make -C tests/riotboot buildtest-indocker
</summary>

```
make: Entering directory '/home/francisco/workspace/RIOT/tests/riotboot'
Launching build container using image "riot/riotbuild:latest".
mkdir -p /home/francisco/workspace/RIOT/build
docker run --rm -t -u "$(id -u)" \
    -v '/usr/share/zoneinfo/Europe/Paris:/etc/localtime:ro' -v '/home/francisco/workspace/RIOT:/data/riotbuild/riotbase' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'     \
    -e 'BOARDS=iotlab-m3 samr21-xpro' \
    -w '/data/riotbuild/riotbase/tests/riotboot/' \
    'riot/riotbuild:latest' make buildtest-indocker  
Building for iotlab-m3 ... success.
Building for samr21-xpro ... success.
make: Leaving directory '/home/francisco/workspace/RIOT/tests/riotboot'

```
</details>

cladmi commented:
I squashed the commits and commit message.

fjmolinas commented:
All green, GO!

cladmi commented:
Thank you for the review! :)
